ansible/roles/test/tasks/advanced-reboot.yml

```diff
@@ -124,14 +142,24 @@
             src: "{{ new_image_location }}"
             dest: "{{ new_image_location }}"
 
-        - name: Install a new SONiC image if requested
-          shell: sonic_installer install -y {{ new_image_location }}
+        - name: Cleanup sonic images that is not current and/or next
+          shell: sonic_installer cleanup -y
           become: true
+          when: cleanup_old_sonic_images|bool
+
+        - name: 'Setup restoring initial image {{ current_sonic_image }}'
+          shell: /bin/true
+          connection: local
           notify:
```

Comment:
>notify [](start = 10, length = 6)

sonic_installer may fails.
So I think we should move this task below sonic_installer, and check sonic_installer return code in when-clause.

Reply:
This my choice. I want to restore system to known good state from this point on regardless if install succeeded or not. Otherwise, moving the task after install task would be sufficient since install failure would abort the job.

Reply:
Are you ok with the choice?